### PR TITLE
Gomuko Game

### PR DESCRIPTION
**Pull request by fadiatamny**
This pull request contains the working base version of Gomuko with all its modes.

this version of the game has issues with the Ai service and missing some testing because of conflicts and issues converting some code to work with the current state of the project.

### CODE REVIEW

**Comment by yoav-zibin** [comment]
Some checks were not successful
1 failing and 1 cancelled checks

Can you fix?

File: .eslintrc.js [hunk 1]
```diff
@@ -77,5 +76,8 @@ module.exports = {
     '@typescript-eslint/no-non-null-assertion': 'error',
     'jest/no-disabled-tests': 'error',
     'jest/no-commented-out-tests': 'error',
+    'new-cap': 'off',
+    'no-self-assign': 'off',
+    'react/jsx-no-undef': 'off',
```
Comment (yoav-zibin) [inline .eslintrc.js]: please revert.
If you want, you can always disable using a comment like:

// eslint-disable-next-line @typescript-eslint/no-explicit-any

**Comment by fadiatamny** [comment]
> Some checks were not successful
> 1 failing and 1 cancelled checks
> 
> Can you fix?

looking into the matter and fixing.

File: gomuko/gameLogic.ts [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+      previousPosJustCapturedForKo.col === col
+    ) {
+      throw Error('KO!');
+    }
+    // bad delta should automatically throw error
+    boardAfterMove[row][col] = turnIndexBeforeMove === 0 ? 'B' : 'W';
+    passesAfterMove = 0; // if a move is made, passes is reset
+    // evaluate board
+    // Check for win -- NEED FIX, why the hell it goes through all positions on board and says it's W?
+    // WTF is this bull? need fix before passNplay is active. for rest just play as black or whatever
```
Comment (yoav-zibin) [inline gomuko/gameLogic.ts]: please remove comment and fix the code :)

File: package-lock.json [hunk 3]
```diff
@@ -19,7 +19,6 @@
       "version": "7.9.0",
       "resolved": "https://registry.npmjs.org/@babel/core/-/core-7.9.0.tgz",
       "integrity": "sha512-kWc7L0fw1xwvI0zi8OKVBuxRVefwGOrKSQMvrQ3dW+bIIavBY3/NpXmpjMy7bQnLgwgzWQZ8TlM57YHpHNHz4w==",
-      "dev": true,
```
Comment (yoav-zibin) [inline package-lock.json]: revert this file

File: gomuko/__tests__/gameLogic.test.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      ['', '', '', '', '', '', '', '', ''],
+      ['', '', '', '', '', '', '', '', ''],
+      ['', '', '', '', '', '', '', '', ''],
+      ['', '', '', '', '', '', '', '', ''],
+    ];
+
+    expect(deepEquals(board, expectedBoard)).toBe(true);
+  });
+
+  // it('Initial move', function () {
```
Comment (yoav-zibin) [inline gomuko/__tests__/gameLogic.test.ts]: add a LOT more unit tests...

**Review by yoav-zibin — CHANGES_REQUESTED** [review]

File: gomuko/__tests__/gameLogic.test.ts (hunk 4 above)
Comment (fadiatamny) [reply gomuko/__tests__/gameLogic.test.ts]: understandable its is as we noted in class the last thing we need to check while fixing the ai again. 
will do once the ai is fixed and all is well.

**Comment by sharon-cohen** [comment]
The code looks good. I loved the design of the code . As one who does not know the role game I am missing more comments

**Pull request closed without merging**